<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="0" total="12" passed="12">
  <reporter-output>
  </reporter-output>
  <suite name="Default suite" duration-ms="751" started-at="2012-05-31T12:15:13Z" finished-at="2012-05-31T12:15:14Z">
    <groups>
    </groups>
    <test name="Default test" duration-ms="751" started-at="2012-05-31T12:15:13Z" finished-at="2012-05-31T12:15:14Z">
      <class name="frameworktests.TestFramework">
        <test-method status="PASS" signature="setup()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="setup" is-config="true" duration-ms="116" started-at="2012-05-31T12:15:13Z" finished-at="2012-05-31T12:15:13Z">
        </test-method> <!-- setup -->
        <test-method status="PASS" signature="testDBConnection()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testDBConnection" duration-ms="554" started-at="2012-05-31T12:15:13Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testDBConnection -->
        <test-method status="PASS" signature="testGetAMSJsonFiles()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testGetAMSJsonFiles" duration-ms="27" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testGetAMSJsonFiles -->
        <test-method status="PASS" signature="testGetBTRJsonFiles()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testGetBTRJsonFiles" duration-ms="2" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testGetBTRJsonFiles -->
        <test-method status="PASS" signature="testGetInvalidAttributeValue()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testGetInvalidAttributeValue" duration-ms="1" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
          <exception class="com.webdriver.qa.automation.framework.exceptions.InvalidAttributeException">
            <message>
              <![CDATA[Could not locate the following attribute: I_AM_NOT_HERE]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.webdriver.qa.automation.framework.exceptions.InvalidAttributeException: Could not locate the following attribute: I_AM_NOT_HERE
	at com.webdriver.qa.automation.framework.testcase.TestConfig.getValue(TestConfig.java:21)
	at frameworktests.TestFramework.testGetInvalidAttributeValue(TestFramework.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:702)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:894)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1219)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:768)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:87)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
	at org.testng.TestNG.run(TestNG.java:1022)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:109)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:202)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:173)
]]>
            </full-stacktrace>
          </exception> <!-- com.webdriver.qa.automation.framework.exceptions.InvalidAttributeException -->
        </test-method> <!-- testGetInvalidAttributeValue -->
        <test-method status="PASS" signature="testGetSuccessfulAttributeValue()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testGetSuccessfulAttributeValue" duration-ms="1" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testGetSuccessfulAttributeValue -->
        <test-method status="PASS" signature="testInvalidBrowserType()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testInvalidBrowserType" duration-ms="2" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
          <exception class="com.webdriver.qa.automation.framework.exceptions.InvalidBrowserTypeException">
            <message>
              <![CDATA[The following browser type was not set properly: "THIS IS NOT A BROWSER!"]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.webdriver.qa.automation.framework.exceptions.InvalidBrowserTypeException: The following browser type was not set properly: "THIS IS NOT A BROWSER!"
	at com.webdriver.qa.automation.framework.Framework.loadFramework(Framework.java:54)
	at frameworktests.TestFramework.testInvalidBrowserType(TestFramework.java:42)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:702)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:894)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1219)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:768)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:87)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
	at org.testng.TestNG.run(TestNG.java:1022)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:109)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:202)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:173)
]]>
            </full-stacktrace>
          </exception> <!-- com.webdriver.qa.automation.framework.exceptions.InvalidBrowserTypeException -->
        </test-method> <!-- testInvalidBrowserType -->
        <test-method status="PASS" signature="testInvalidJsonEnvironment()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testInvalidJsonEnvironment" duration-ms="2" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
          <exception class="com.webdriver.qa.automation.framework.exceptions.InvalidEnvironmentException">
            <message>
              <![CDATA[The following environment was not set properly: "THIS IS NOT AN ENVIRONMENT!"]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.webdriver.qa.automation.framework.exceptions.InvalidEnvironmentException: The following environment was not set properly: "THIS IS NOT AN ENVIRONMENT!"
	at com.webdriver.qa.automation.framework.Framework.loadFramework(Framework.java:58)
	at frameworktests.TestFramework.testInvalidJsonEnvironment(TestFramework.java:52)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:702)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:894)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1219)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:768)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:87)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
	at org.testng.TestNG.run(TestNG.java:1022)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:109)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:202)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:173)
]]>
            </full-stacktrace>
          </exception> <!-- com.webdriver.qa.automation.framework.exceptions.InvalidEnvironmentException -->
        </test-method> <!-- testInvalidJsonEnvironment -->
        <test-method status="PASS" signature="testNonExistingProject()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testNonExistingProject" duration-ms="3" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
          <exception class="com.webdriver.qa.automation.framework.exceptions.TestProjectNotFoundException">
            <message>
              <![CDATA[Cannot locate the following project: null
 Available Projects: [BTR, AMS, FRAMEWORK]]]>
            </message>
            <full-stacktrace>
              <![CDATA[com.webdriver.qa.automation.framework.exceptions.TestProjectNotFoundException: Cannot locate the following project: null
 Available Projects: [BTR, AMS, FRAMEWORK]
	at com.webdriver.qa.automation.framework.Framework.loadFramework(Framework.java:62)
	at frameworktests.TestFramework.testNonExistingProject(TestFramework.java:67)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:702)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:894)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1219)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:768)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:87)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
	at org.testng.TestNG.run(TestNG.java:1022)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:109)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:202)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:173)
]]>
            </full-stacktrace>
          </exception> <!-- com.webdriver.qa.automation.framework.exceptions.TestProjectNotFoundException -->
        </test-method> <!-- testNonExistingProject -->
        <test-method status="PASS" signature="testProperJsonBrowserType()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testProperJsonBrowserType" duration-ms="1" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testProperJsonBrowserType -->
        <test-method status="PASS" signature="testProperJsonEnvironment()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testProperJsonEnvironment" duration-ms="1" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testProperJsonEnvironment -->
        <test-method status="PASS" signature="testProperJsonImplicitWaitTime()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testProperJsonImplicitWaitTime" duration-ms="1" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testProperJsonImplicitWaitTime -->
        <test-method status="PASS" signature="testProperJsonProjectType()[pri:0, instance:frameworktests.TestFramework@31b5998f]" name="testProperJsonProjectType" duration-ms="1" started-at="2012-05-31T12:15:14Z" finished-at="2012-05-31T12:15:14Z">
        </test-method> <!-- testProperJsonProjectType -->
      </class> <!-- frameworktests.TestFramework -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
